pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -2211,6 +2658,85 @@ func verifyBindStatusAvailableWithOnePlacement(binding *placementv1beta1.Cluster
 	}, timeout, interval).Should(BeEmpty(), fmt.Sprintf("binding(%s) mismatch (-want +got)", binding.Name))
 }
 
+// TO-DO (chenyu1): consider refactoring the related code to reduce duplication.
+func workApplyStrategyUpdatedActual(workNamespace, workName string, wantApplyStrategy *placementv1beta1.ApplyStrategy) func() error {
+	return func() error {
+		work := &placementv1beta1.Work{}
+		if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: workNamespace, Name: workName}, work); err != nil {
+			return fmt.Errorf("failed to retrieve work %s: %w", workName, err)
+		}
+
+		gotApplyStrategy := work.Spec.ApplyStrategy
+		if diff := cmp.Diff(gotApplyStrategy, wantApplyStrategy); diff != "" {
+			return fmt.Errorf("work apply strategy mismatches (-got, +want):\n%s", diff)
+		}
+		return nil
+	}
+}
+
+func bindingStatusUpdatedActual(
```

Comment:
looks like this can be used to replace a bunch of "verifyBindStatusXXX" function? If so, please add a TODO

Reply:
although the full comparison mode looks more verbose

Reply:
I will add a TO-DO comment and think about the best way to refactor the code 👌.